root/utility/tableColumns.js

```diff
@@ -10,27 +10,82 @@
 import * as React from 'react';
 import type {ColumnOptions} from 'react-table';
 
+import SortableTableHeader from '../components/SortableTableHeader';
+import DescriptiveLink
+  from '../static/scripts/common/components/DescriptiveLink';
 import EntityLink from '../static/scripts/common/components/EntityLink';
 import yesNo from '../static/scripts/common/utility/yesNo';
 
-export function defineNameColumn(
+export function defineCheckboxColumn<T>(
+  name: string,
+): ColumnOptions<EntityRoleT<T>, number> {
+  return {
+    Cell: ({cell: {value}}) => (
+      <input
+        name={name}
+        type="checkbox"
+        value={value}
+      />
+    ),
+    Header: <input type="checkbox" />,
+    accessor: 'id',
+    className: 'checkbox-cell',
+    id: 'checkbox',
+  };
+}
+
+export function defineNameColumn<T: CoreEntityT | CollectionT>(
+  descriptive: boolean,
+  order: ?string,
+  sortable: ?boolean,
   title: string,
-): ColumnOptions<CoreEntityT | CollectionT, string> {
+): ColumnOptions<T, string> {
   return {
-    Cell: ({row: {original}}) => (
-      <EntityLink entity={original} />
+    Cell: ({row: {original}}) => (descriptive
+      // $FlowFixMe
+      ? <DescriptiveLink entity={original} />
```

Comment:
This is due to `descriptive` argument being redundant with flow type. Maybe there should be a separate function for `CollectionT`, or maybe `DescriptiveLink` can be used with collection instead of `EntityLink` below.